### ISSUE TITLE
Input's form attribute and document submit handler produce an IE8 error

### ISSUE THREAD

**Issue by Igor10k** [jquery/jquery#2332]
When you use an input with `form` attribute while also handling the document's `submit` event you get an error in IE8.

Here's an easy way to reproduce it.

```html
<html>
<body>
  <input type="text" form="test">
  <form action="" id="test"></form>
  <script src="//code.jquery.com/jquery-1.11.3.min.js"></script>
  <script>
    $(document).on('submit', 'form', function () {});
  </script>
</body>
</html>
```

When you click the input in IE8 you get this error.

![screen shot 2015-05-19 at 16 10 42](https://cloud.githubusercontent.com/assets/1244342/7703590/a24b82d2-fe41-11e4-891d-ee385b0faa23.png)

Here's a jsfiddle http://jsfiddle.net/1g19xt3p/1/
And since jsfiddle doesn't work in IE8 here's a direct link to the result https://jsfiddle.net/1g19xt3p/1/embedded/result/

A small research showed that here's the code responsible
```js
jQuery.event.special.submit = {
  setup: function() {
    // skipped
    jQuery.event.add( this, "click._submit keypress._submit", function( e ) {
      var elem = e.target,
        form = jQuery.nodeName( elem, "input" ) || jQuery.nodeName( elem, "button" ) ? elem.form : undefined;
      if ( form && !jQuery._data( form, "submitBubbles" ) ) {
        // skipped
      }
    });
  },
  // skipped
```

The thing is `elem.form` returns the `form` attribute value in IE8 that is a `String` and jQuery can't get the data out of a `String`.

**Comment by dmethvin** [comment]
See [trac-12717](http://bugs.jquery.com/ticket/12717) and [trac-14126](http://bugs.jquery.com/ticket/14126). Also note that Microsoft will [stop supporting IE8 in January 2016](http://blogs.msdn.com/b/ie/archive/2014/08/07/stay-up-to-date-with-internet-explorer.aspx) so it's really not worth a lot of additional IE8 heroics at this point.

**Comment by Igor10k** [comment]
@dmethvin 

I see your point, but I still think that it wouldn't hurt to change
```js
if ( form && !jQuery._data( form, "submitBubbles" ) ) {
```
to
```js
if ( form && typeof form !== "string" && !jQuery._data( form, "submitBubbles" ) ) {
```
or maybe even
```js
if ( typeof form == "object" && !jQuery._data( form, "submitBubbles" ) ) {
```

It won't make the `form` attribute suddenly work in IE8. But it will allow to create such solutions without hacking into `jQuery.event.special.submit`. Or am I missing something?

**Comment by dmethvin** [comment]
That might prevent an error but it won't make form elements work that have a string `form` property though. I'm not a big fan of silent failure. Really, IE8 doesn't support HTML5 form features and anyone who wants to support IE8 will need to deal with it.

**Comment by Igor10k** [comment]
My reason is that when I decide to go with using `form` attribute I know that I'll have to implement some solution to make it work in oldIE. What I don't know at that moment is that jQuery will throw errors. So instead of just doing the planned polyfill I have to dig in the jQuery code to understand where the error comes from.

**Comment by gibson042** [comment]
We should replace `elem.form` with `jQuery.prop( elem, "form" )` so a propHook can fix it. This is exactly analogous to https://github.com/jquery/jquery/commit/ae215fdcf843862dbddbdfd68117ac2926e35a42#diff-89c55bb1a12d31cb4eb5d007633dc33bR12 .

**Comment by dmethvin** [comment]
Oh, true, I didn't even think about using a propHook!

**Comment by mzgol** [comment]
@gibson042 
> We should replace `elem.form` with `jQuery.prop( elem, "form" )` so a propHook can fix it.

Let me make sure if I understand you correctly - do you suggest just to switch `elem.form` to `jQuery.prop( elem, "form" )` so that jQuery users can define a proper `propHook` if they need IE8 support and they need to workaround this issue?

**Comment by gibson042** [comment]
Correct.

**Comment by mzgol** [comment]
Fixed in ead83b9c8a6b751f31f0197ff0364e95a1ca24cd.

Note that we only use the `propHook`, to make it work with IE 8 you'd have to register one that polyfills HTML5 form features.
